crates/compilers/src/preprocessor/data.rs

```diff
@@ -0,0 +1,209 @@
+use crate::preprocessor::SourceMapLocation;
+use foundry_compilers_artifacts::{Source, Sources};
+use itertools::Itertools;
+use solar_parse::interface::{Session, SourceMap};
+use solar_sema::{
+    hir::{Contract, ContractId, Hir},
+    interface::source_map::FileName,
+};
+use std::{
+    collections::{BTreeMap, HashSet},
+    path::{Path, PathBuf},
+};
+
+/// Keeps data about project contracts definitions referenced from tests and scripts.
+/// HIR id -> Contract data definition mapping.
+pub type PreprocessorData = BTreeMap<u32, ContractData>;
+
+/// Collects preprocessor data from referenced contracts.
+pub fn collect_preprocessor_data(
```

Comment:
please avoid making stuff pub as much as possible if it's not needed outside of this crate, we don't want to expose implementation details.

Reply:
thanks, I overlooked this, I changed the visibility in scope of crate with https://github.com/foundry-rs/compilers/pull/252/commits/f4a109d53bfe97bba287335404c2f5cc20e7dbf6